### PR TITLE
Issue #13345: Enable examples tests for MethodCountCheck

### PR DESCRIPTION
**Pull request by AmitKumarDeoghoria**
### Title: Enable examples tests for MethodCountCheck
**Related Issue: #13345** 

**Summary**
This pull request aims to enable example tests for the **MethodCountCheck** in the project. The implementation focuses on ensuring that MethodCount checks are correctly validated through comprehensive test cases, improving overall code reliability and quality.

**Key Changes Made**

- Conversion of txt file to Java File
- Enable the test cases by adding expected violations in the expected array for all example tests
- Made corresponding changes in the xml and xml.template file

### CODE REVIEW

**Comment by AmitKumarDeoghoria** [comment]
GitHub, generate website

**Comment by AmitKumarDeoghoria** [comment]
GitHub, generate website

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/sizes/methodcount/Example1.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    <module name="MethodCount"/>
+  </module>
+</module>
+*/
+package com.puppycrawl.tools.checkstyle.checks.sizes.methodcount;
+
+// xdoc section -- start
+class Example1 {
+
+  public void outerMethod1(int i){}
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/sizes/methodcount/Example1.java:13]: Please put space before each `{`. In all examples.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/sizes/methodcount/Example1.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  public void outerMethod2(){}
+  public void outerMethod3(String str){}
+
+  private void outerMethod4(){
+    Runnable r = (new Runnable() {
+      public void run() {} // NOT counted towards ExampleClass
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/sizes/methodcount/Example1.java:19]: towards Example1


Fix in each example.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/sizes/methodcount/Example1.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    });
+  }
+
+  private void outerMethod5(int i){}
+  void outerMethod6(int i , int j){}
+
+  public static class InnerExample{
+    public void InnerMethod1(){} // NOT counted towards ExampleClass,
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/sizes/methodcount/Example1.java:27]: `innerMethod1()`

Same for next method

**Review by romani — CHANGES_REQUESTED** [review]
Items

**Comment by romani** [comment]
GitHub, generate website

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/sizes/methodcount/Example1.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    Runnable r = (new Runnable() {
+      public void run() {} // NOT counted towards Example1
+    });
+  }
+
+  private void outerMethod5(int i) {}
+  void outerMethod6(int i , int j) {}
+
+  public static class InnerExample{
+    public void innerMethod1() {} // NOT counted towards Example1,
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/sizes/methodcount/Example1.java:27]: please remove traililng `,`  in all comment

**Review by romani — CHANGES_REQUESTED** [review]
last

**Comment by AmitKumarDeoghoria** [comment]
GitHub, generate website

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/sizes/methodcount/Example1.java (hunk 2 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/sizes/methodcount/Example1.java]: Done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/sizes/methodcount/Example1.java (hunk 3 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/sizes/methodcount/Example1.java]: Done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/sizes/methodcount/Example1.java (hunk 4 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/sizes/methodcount/Example1.java]: Done

**Review by romani — APPROVED** [review]
thanks a lot!!!

**Pull request merged**